lib/Conversion/HWToBTOR2/HWToBTOR2.cpp

```diff
@@ -458,20 +465,14 @@ struct ConvertHWToBTOR2Pass
     return width;
   }
 
-  // Generates the transitions required to finalize the register to state
-  // transition system conversion
-  void finalizeRegVisit(Operation *op) {
-    // Check the register type (done to support non-firrtl registers as well)
-    auto reg = cast<seq::FirRegOp>(op);
-
-    // Generate the reset condition (for sync & async resets)
-    // We assume for now that the reset value is always 0
-    size_t width = hw::getBitWidth(reg.getType());
+  // Emits the btor2 instructions required to define a state transition system
+  // for the given register
+  void finalizeGenRegVisit(Operation *reg, size_t width) {
     genSort("bitvec", width);
 
     // Extract the `next` operation for each register (used to define the
     // transition). We need to check if next is a port to avoid nullptrs
-    Value next = reg.getNext();
+    Value next = reg->getOperand(0);
```

Comment:
Right, I wanted to avoid code duplication, but I guess it's best to have it be more readable and more reliable in case the ordering of operands changes for some reason.